### ISSUE TITLE
Fatal error: Facebook contains 6 abstract methods

### ISSUE THREAD

**Issue by PhantomWatson** [thephpleague/oauth2-facebook#19]
I'm getting the following error when trying to use the latest version (c7ea7041b597a4d215abe49cf82dbede6eaed1c1) of this library:

```
Error: Class League\OAuth2\Client\Provider\Facebook contains 6 abstract methods 
and must therefore be declared abstract or implement the remaining methods 
(League\OAuth2\Client\Provider\AbstractProvider::getBaseAuthorizationUrl, 
League\OAuth2\Client\Provider\AbstractProvider::getBaseAccessTokenUrl, 
League\OAuth2\Client\Provider\AbstractProvider::getResourceOwnerDetailsUrl, ...)
File /.../league/oauth2-client/src/Provider/Facebook.php
Line: 8 
```

**Comment by SammyK** [comment]
Can you post the snippet of code that's causing this error? I just did a fresh install & copy+pasted the example from the readme and it's working as expected.

**Comment by PhantomWatson** [comment]
```
return new \League\OAuth2\Client\Provider\Facebook([
    'clientId' => '...',
    'clientSecret' => '...',
    'redirectUri' => '...',
    'graphApiVersion' => 'v2.5'
]);
```

**Comment by SammyK** [comment]
That's correct so it should work. It seems something wonky is going on with either the autoloader or there's some weird alias of `Facebook` to the `AbstractProvider`. Have you tried doing `composer dump-autoload`? Or trying to do a composer update on the package again?

**Comment by PhantomWatson** [comment]
Yep. I ran `composer dump-autoload`, I deleted the package and ran `composer update` again, and still the same results. Oddly, I'm not getting the error when I run it on localhost, only on my production server, which various FTP clients are telling me is hosting 100% identical code.

I wouldn't expect this to matter, but locally, I'm running PHP 7, and on the production server I'm running PHP 5.5.33.

**Comment by SammyK** [comment]
Are you also running composer install on your production server? Sounds like an environment-specific issue.

**Comment by PhantomWatson** [comment]
Nah, just running composer locally and syncing with FTP.

**Comment by SammyK** [comment]
Gotcha. Well sounds like something's getting twisted around from local to FTP'ing to production. I'll close this for now but feel free to post any more info that might help us narrow this down to a bug with the lib. :)

**Comment by PhantomWatson** [comment]
Oh, I figured it out! I noticed that the error message referred to `/.../league/oauth2-client/src/Provider/Facebook.php`, which isn't actually a file in the current version of this library. Turns out composer had downloaded an older version of `oath2-client` in an earlier attempt because I wasn't running a sufficiently-recent version of PHP locally. I upgraded to PHP 7 and got the latest version of `oath2-facebook` and `oath2-client`, but by then I had already uploaded the older version to production and it was incorrectly reading from files like `/.../league/oauth2-client/src/Provider/Facebook.php` that should have been deleted.

Sorry for the trouble!
